agent/bench-scripts/pbench-uperf

```diff
@@ -181,280 +182,295 @@ function gen_xml {
 		;;
 		-m|--message-sizes)
-		shift
-		if [ -n "$1" ]; then
-			message_sizes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				message_sizes="$1"
+				shift
+			fi
 		;;
 		-r|--runtime)
-		shift
-		if [ -n "$1" ]; then
-			runtime="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				runtime="$1"
+				shift
+			fi
 		;;
 		--log-response-times)
-		shift
-		if [ -n "$1" ]; then
-			log_response_times="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				log_response_times="$1"
+				shift
+			fi
 		;;
 		-p|--protocols)
-		shift
-		if [ -n "$1" ]; then
-			protocols="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				protocols="$1"
+				shift
+			fi
 		;;
 		-C|--client|--clients)
-		shift
-		if [ -n "$1" ]; then
-			clients="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				clients="$1"
+				shift
+			fi
 		;;
 		-S|--server|--servers)
-		shift
-		if [ -n "$1" ]; then
-			servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				servers="$1"
+				shift
+			fi
 		;;
 		--vsock-server|--vsock-servers)
-		shift
-		if [ -n "$1" ]; then
-			vsock_servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				vsock_servers="$1"
+				shift
+			fi
 		;;
 		-c|--config)
-		shift
-		if [ -n "$1" ]; then
-			config="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				config="$1"
+				shift
+			fi
 		;;
 		--start-iteration-num)
-		shift
-		if [ -n "$1" ]; then
-			start_iteration_num=$1
-			shift
-		fi
+			if [ -n "$1" ]; then
+				start_iteration_num=$1
+				shift
+			fi
 		;;
 		--client-node|--client-nodes)
-		shift
-		if [ -n "$1" ]; then
-			client_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				client_nodes="$1"
+				shift
+			fi
 		;;
 		--server-node|--server-nodes)
-		shift
-		if [ -n "$1" ]; then
-			server_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				server_nodes="$1"
+				shift
+			fi
 		;;
 		--sysinfo)
-		shift;
-		if [ -n "$1" ]; then
-			sysinfo="$1"
-			shift;
-		fi
+			if [ -n "$1" ]; then
+				sysinfo="$1"
+				shift
+			fi
 		;;
 		--)
-		shift
-		break
+			break
 		;;
 		*)
-		error_log "[$script_name] bad option, \"$1 $2\""
-		exit 1
-		break
+			error_log "[$script_name] bad option, \"${arg} ${1}\""
+			usage >&2
+			exit 1
 		;;
 	esac
 done
 verify_common_bench_script_options $tool_group $sysinfo
 
-# Verify the number of clients and servers match.
-if [ ! -z "$clients" -a "$postprocess_only" != "y" ]; then
+clients=${clients:-${clients_def}}
+servers=${servers:-${servers_def}}
+
+if [[ "${postprocess_only}" != "y" ]]; then
+	# Boolean value recording if local clients or servers are specified.
+	let check_local=0
+
+	# Verify that the number of clients, servers, and optional vsock servers
+	#  match.
 	let c_cnt=0
-	for client in `echo $clients | sed -e s/,/" "/g`; do
-		let c_cnt=c_cnt+1
+	for client in ${clients//,/ }; do
+		(( c_cnt++ ))
+		if ! is_remote "${client}"; then
+			check_local=1
+		fi
 	done
+	if [[ ${c_cnt} -eq 0 ]]; then
+		error_log "[${script_name}] INTERNAL ERROR - no clients found"
+		exit 1
+	fi
```

Comment:
It's hard to track the motion of comments across rebases; was this the same redundant check I pointed out before, or a different one? Since `clients` and `servers` are both defaulted to `127.0.0.1`, `c_cnt` can't be 0. Right?

Reply:
Sort of.  As part of other work, I had noticed this needed to be changed.  You can see it via a `git blame` on the file from the PR branch: https://github.com/portante/pbench/blame/refactor-pbench-uperf/agent/bench-scripts/pbench-uperf

I had fixed this as part of the `is_remote` commit: https://github.com/portante/pbench/commit/39148a0aa9e0e1b8425d835a29c29bb81ea202ef